### DISCUSSION TITLE
User credentials under Linux

### DISCUSSION THREAD

**Discussion by halloleo** [genotrance/px#142] [Q&A] [answered]
Firstly, thanks heaps for developing `px` - it might solve big troubles for me!

Secondly, if I want to use `px` on Linux (RHEL 8), how do I provide the user credentials (particularly the password)? My Linux system doesn't "knows" the credentials for the NTLM proxy.

Thanks for any pinters!

### Background:

In the past, on Linux RHEL 6, I used [cntlm](http://cntlm.sourceforge.net/) where I provide the username and the  hashed password in a config file. This worked great, but I cannot get cntlm running under RHEL 8...

**Comment by genotrance** [comment, marked as answer]
Right now, the recommendation is to use [keyring](https://pypi.org/project/keyring/) which is a standard password storage mechanism across operating systems. This is documented in the [README](https://github.com/genotrance/px#credentials).

It is super easy to set up on Windows and I've had good success on Ubuntu and Alpine Linux. However on Red Hat based systems, I've had some trouble if there is no GUI, for example in headless or docker setups. The recommendation is to use the [keyring_jeepney](https://pypi.org/project/keyring_jeepney/) package in such cases.

Please let me know if you have any trouble, will be happy to support getting this working on Linux.

**Reply by halloleo** [reply]
Thanks @genotrance for chiming in!

Before installing more stuff I would like to _test_ whether `px` can work in our network set-up. Is there a way to provide for testing the credentials on the command line?

**Reply by genotrance** [reply]
Right now, you would need to get Px running on your setup either using pip or the binary releases if you are running a supported distro.

In the hopefully near future, you can be confident that Px will work anywhere that curl does since I am close to releasing v0.7.0 that replaces the current backend with libcurl. You would simply run:

`curl --proxy-anyauth --proxy-user "user:password" -v http://google.com`

If it works, you will know Px can as well.
